magenta/music/melodies_lib.py

```diff
@@ -857,38 +892,216 @@ def squash_and_encode(self, melody):
 
 
 class OneHotMelodyEncoderDecoder(MelodyEncoderDecoder):
```

Comment:
While you're refactoring, one nit I've always had is that "EncoderDecoder" is actually a poor name choice because it conflicts with the whole concept of EncoderDecoder from Seq2Seq and Autoencoders. This looks much more like "Preprocessing" to me. Another common term is "DataLoader" or "DataReader"